smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableApiConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.traits.tagging;
+
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.node.ToNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.ToSmithyBuilder;
+
+/**
+ * Structure representing the configuration of resource specific tagging APIs.
+ */
+public final class TaggableApiConfig implements FromSourceLocation, ToNode, ToSmithyBuilder<TaggableApiConfig> {
+    private final ShapeId tagApi;
+    private final ShapeId untagApi;
+    private final ShapeId listTagsApi;
+    private final SourceLocation sourceLocation;
+
+    private TaggableApiConfig(Builder builder) {
+        tagApi = builder.tagApi;
+        untagApi = builder.untagApi;
+        listTagsApi = builder.listTagsApi;
+        sourceLocation = builder.sourceLocation != null
```

Comment:
Why not just default this property to SourceLocation.none() in the builder and remove this logic?

Reply:
I don't feel too strongly about it, but this approach removes the possibility of source location being set to null even if that's what the caller did with the builder. It doesn't read as clean and I don't think it's a serious concern so I'll change it.

Reply:
Just use `Objects.requireNonNull` in the builder